engine/src/main/java/org/camunda/bpm/engine/impl/cmd/AbstractCorrelateMessageCmd.java

```diff
@@ -112,7 +115,11 @@ protected MessageCorrelationResultImpl createMessageCorrelationResult(final Comm
     MessageCorrelationResultImpl resultWithVariables = new MessageCorrelationResultImpl(handlerResult);
     if (MessageCorrelationResultType.Execution.equals(handlerResult.getResultType())) {
       ExecutionEntity execution = findProcessInstanceExecution(commandContext, handlerResult);
-      if (variablesEnabled && execution != null) {
+
+      ProcessInstance processInstance = execution.getProcessInstance();
+      resultWithVariables.setProcessInstance(processInstance);
```

Comment:
Hint for the reviewer: This line is not directly associated with the feature but it is necessary to set the process instance for the message correlation, otherwise the operation log will not contain the process instance. I believe this was not implemented by mistake.